.github/workflows/staging-messaging-test.yml

```diff
@@ -0,0 +1,39 @@
+name: Staging - TON2EVM messaging test
+
+on:
+  schedule:
+    # Runs at minute 0, 15, 30, and 45 of every hour
+    - cron: "*/15 * * * *"
+
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+    branches:
+      - main
+
+jobs:
+  messaging-test:
+    if: |
+      github.event_name == 'schedule' ||
+      (github.event_name == 'pull_request' &&
+       contains(github.event.pull_request.labels.*.name, 'run-staging-ton2evm-test'))
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@02a151ada4993995686f9ed4f1be7cfbb229e56f # v31
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - name: Run TON2EVM messaging test
+        run: |
+          pushd staging-messaging-test
+          nix develop -c ./run_test.sh \
+            "${{ secrets.STAGING_TON_ROUTER_ADDRESS }}" \
+            "${{ secrets.STAGING_SEPOLIA_DEST_CHAIN_SELECTOR }}" \
```

Comment:
Chain selectors are not secrets https://github.com/smartcontractkit/chain-selectors/blob/main/selectors.yml#L270